torch/_inductor/codegen/cpp.py

```diff
@@ -1872,6 +2065,22 @@ def broadcast(self, scalar_var: CppCSEVariable):
         vec_var.is_vec = True
         return vec_var
 
+    def arange(
+        self, index: Union[sympy.Expr, CppCSEVariable], stride: sympy.Symbol
+    ) -> CppCSEVariable:
+        if isinstance(index, sympy.Expr):
+            index = cexpr(index)
+        else:
+            assert isinstance(index, CppCSEVariable)
+            assert not index.is_vec
+        csevar = self.cse.generate(
+            self.compute, f"at::vec::Vectorized<int32_t>::arange({index}, {stride})"
```

Comment:
This works because we know that if we needed to perform the indexing with `int64` we wouldn't be on the vectorization path, right?

Reply:
Explained.